pandas/tests/frame/methods/test_swapaxes.py

```diff
@@ -8,22 +8,30 @@
 class TestSwapAxes:
     def test_swapaxes(self):
         df = DataFrame(np.random.randn(10, 5))
-        tm.assert_frame_equal(df.T, df.swapaxes(0, 1))
-        tm.assert_frame_equal(df.T, df.swapaxes(1, 0))
+        msg = "'DataFrame.swapaxes' is deprecated"
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            tm.assert_frame_equal(df.T, df.swapaxes(0, 1))
+            tm.assert_frame_equal(df.T, df.swapaxes(1, 0))
 
     def test_swapaxes_noop(self):
         df = DataFrame(np.random.randn(10, 5))
-        tm.assert_frame_equal(df, df.swapaxes(0, 0))
+        msg = "'DataFrame.swapaxes' is deprecated"
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            tm.assert_frame_equal(df, df.swapaxes(0, 0))
 
     def test_swapaxes_invalid_axis(self):
         df = DataFrame(np.random.randn(10, 5))
-        msg = "No axis named 2 for object type DataFrame"
-        with pytest.raises(ValueError, match=msg):
-            df.swapaxes(2, 5)
+        msg = "'DataFrame.swapaxes' is deprecated"
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            msg = "No axis named 2 for object type DataFrame"
+            with pytest.raises(ValueError, match=msg):
+                df.swapaxes(2, 5)
```

Comment:
Not sure if nested blocks of code are better expressed. (`pytest` may break pre-commit)